microservices/Keli-Katheya-User-Registration/src/main/java/com/kelikatheya/microservices/userRegistration/security/AuthenticationFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.kelikatheya.microservices.userRegistration.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.kelikatheya.microservices.userRegistration.entity.UserEntity;
+import com.kelikatheya.microservices.userRegistration.requestModels.UserLoginRequest;
+import com.kelikatheya.microservices.userRegistration.services.UserService;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+
+public class AuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private Environment environment;
+
+    public AuthenticationFilter(UserService userService , Environment environment , AuthenticationManager authenticationManager){
+        this.userService = userService;
+        this.environment = environment;
+        super.setAuthenticationManager(authenticationManager);
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest httpServletRequest , HttpServletResponse httpServletResponse) {
+        try {
+            UserLoginRequest creds = new ObjectMapper().readValue(httpServletRequest.getInputStream(), UserLoginRequest.class);
+            return getAuthenticationManager().authenticate(
+                    new UsernamePasswordAuthenticationToken(creds.getEmail() , creds.getPassword() , new ArrayList<>())
+            );
+        } catch (IOException e) {
+            throw new RuntimeException();
```

Comment:
We need exception handlers to give proper response if something goes wrong

Reply:
DO we need to consider that now or can we take up later?

Reply:
If there are errors, then we will have to show generic errors on android end. and do refactoring later on when we decide its time. So better do it now only